org.eclipse.jdt.ls.core/src/org/eclipse/jdt/internal/corext/refactoring/structure/constraints/ConditionalTypeConstraint.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2005, 2011 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.corext.refactoring.structure.constraints;
+
+import org.eclipse.core.runtime.Assert;
+
+import org.eclipse.jdt.internal.corext.refactoring.typeconstraints2.ConstraintVariable2;
+import org.eclipse.jdt.internal.corext.refactoring.typeconstraints2.ITypeConstraint2;
+
+/**
+ * Type constraint which models conditional expression type constraints.
+ */
+public final class ConditionalTypeConstraint implements ITypeConstraint2 {
```

Comment:
When I look at this class, it seems like it can very easily be moved into `jdt.core.manipulation`. All of its dependants seem to be there. If we could get `ExtractInterfaceProcessor` & `SuperTypeRefactoringProcessor` that would be really good. If the only UI class that is tricky is the usage of `FormatterProfileManager`, I think we may have dealt with that in the past.

Reply:
Yeah I had create a PR in the jdt.ui side to move these types to jdt.core.manipulation: https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/344 before, making these types (org.eclipse.jdt.internal.corext.refactoring.*) available so that we can remove them from this PR.

> If the only UI class that is tricky is the usage of FormatterProfileManager, I think we may have dealt with that in the past.

Could you explain it a little bit more? In https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/344 I just directly use `JavaProject.getOptions(true)` and I'm afraid it can't handle well.